sscanss/editor/designer.py

```diff
@@ -1377,3 +1381,237 @@ def value(self):
 
         # Return updated set of fixed hardware components
         return {self.key: self.hardware_list}
+
+
+class PositioningStacksComponent(QtWidgets.QWidget):
+    """Creates a UI for modifying the positioning stacks component of the instrument description"""
+    def __init__(self):
+        super().__init__()
+
+        self.type = Designer.Component.PositioningStacks
+        self.key = 'positioning_stacks'
+
+        self.json = {}
+        self.folder_path = '.'
+        self.add_new_text = 'Add New...'
+        self.positioning_stack_list = []
+        self.positioners_list = []
+
+        layout = QtWidgets.QGridLayout()
+        self.setLayout(layout)
+
+        # Name field - string, required -- combobox chooses between stacks, and allows renaming
+        self.name_combobox = QtWidgets.QComboBox()
+        self.name_combobox.setEditable(True)
+        layout.addWidget(QtWidgets.QLabel('Name: '), 0, 0)
+        layout.addWidget(self.name_combobox, 0, 1)
+        self.name_validation_label = create_required_label()
+        layout.addWidget(self.name_validation_label, 0, 2)
+
+        # When the positioning stack is changed, connect to a slot that updates the list of positioners in the
+        # component. The "activated" signal is emitted only when the user selects an option (not programmatically)
+        # and is also emitted when the user re-selects the same option.
+        self.name_combobox.activated.connect(lambda: self.updateValue(self.json, self.folder_path))
+
+        # Positioners field - string(s) from list
+        self.positioners_combobox = QtWidgets.QComboBox()
```

Comment:
Can you make it so that a positioner can be added before it is defined similar to the collimator components? so the order in which stuff can be added is flexible.